src/Layer.js

```diff
@@ -334,46 +337,77 @@ export default class Layer {
     }
 
     /**
-     * Change layer visibility to visible
-     *
-     * @memberof carto.Layer
-     * @instance
-     * @api
-     *
-     * @fires updated
+     * Custom Layer API: `onAdd` function
      */
-    show () {
-        this._visible = true;
-        this._integrator.changeVisibility(this);
-        this.requestData();
-        this._fire('updated');
+    onAdd (map, gl) {
+        this.gl = gl;
+        this.map = map;
+        this.renderer = getRenderer(map, gl);
+
+        // Register map event to request data tiles
+        // map.on('move', this.requestData.bind(this));
+
+        // Initialize render layer
+        this._renderLayer.renderer = this.renderer;
+        this._renderLayer.dataframes.forEach(d => d.bind(this.renderer));
+        this._contextInitialize();
+        this.requestMetadata();
```

Comment:
and no requestData too?

Reply:
The first requestData is done in the `update` function called by the constructor.

Reply:
:+1: